rclpy/rclpy/node.py

```diff
@@ -756,6 +756,22 @@ def _set_parameters_atomically(
             self._parameter_event_publisher.publish(parameter_event)
 
         return result
+    
+    def add_on_set_paramters_callback(self,callback:Callable([List[Parameter],SetParametersResult])):
+        """add the callback to list"""
+
+        prevCallBack = self._parameters_callback
+        prevCallBack.insert(0,callback)
```

Comment:
As @ivanpauno suggested, you can do the insert on `self._parameters_callback` instead of using a temporary variable.